daml-lf/spec/daml-lf-1.rst

```diff
@@ -2214,183 +2538,278 @@ as described by the ledger model::
      keys ∈ finite injective map from GlobalKey to cid
 
   Update result
-    ur ::= Ok (v, tr)
-        |  Err v
+    ur ::= Ok (v, tr) ‖ S
+        |  Err t
+    S ::= (st, keys)
 
 
-                                    ┌──────────────────────────────┐
-  Big-step update interpretation    │ u ‖ E₀ ; S₀ ⇓ᵤ ur ‖ E₁ ; S₁  │
-                                    └──────────────────────────────┘
+                                    ┌──────────────┐
+  Big-step update interpretation    │ u ‖ S₀ ⇓ᵤ ur │
+                                    └──────────────┘
 
    —————————————————————————————————————————————————————————————————————— EvUpdPure
```

Comment:
We should make clear that the update interpretation work only on update value (maybe explaining how are handle non value update statement), or add a rule like:

```
u not a value
u ⇓  Ok v
v ‖ E ; (st, keys)   ⇓ᵤ   ur  || S
-----------------------------------------
u ‖ E ; (st, keys)   ⇓ᵤ   ur  || S
```

In the second case I think we could simplify some other rules.

Reply:
I think we should keep the expression evaluation and update interpretation separate, so I'll add something saying that this relation is only defined for update values `u`.

Reply:
:+1:

Reply:
So, the comments before this section already mention this (every `u` in this section has to be a value). But I added a little reminder next to the "box".